crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -854,9 +883,15 @@ impl Object {
     /// `checkpoint_viewed_at` represents the checkpoint sequence number at which this `Object` was
     /// constructed in. This is stored on `Object` so that when viewing that entity's state, it will
     /// be as if it was read at the same checkpoint.
+    ///
+    /// `root_version` represents the version of the root object in some nested ownership, and
+    /// should be propagated from this object's parent, if any.
+    /// If None, then we use [`version_for_dynamic_fields`] to infer a root version to then
```

Comment:
```suggestion
    /// `root_version` represents the version of the root object in some nested chain of dynamic fields. This should typically be left `None`, unless the object(s) being resolved is a dynamic field, or if `root_version` has been explicitly set for this object. 
    /// If None, then we use [`version_for_dynamic_fields`] to infer a root version to then
```


On fetching object, `try_from_stored_history_object` will set the object's `object_version` as `root_version`
- this is passed as `parent_version` to `DynamicField::query` or `::paginate`
- if it's a `df`, it will already be bounded by the `parent_version`
- if it's a `dof`, then when we resolve its `value`, we pass the `root_version` inherited to the fetched `MoveObject`
- when resolving the `MoveObject`'s `DynamicField`, this cycle is perpetuated
However, if we break this chain, then the `root_version` resets until we try to fetch a `DynamicField` again

Reply:
Yeah that sounds correct